graylog2-web-interface/src/components/event-notifications/event-notification-types/HttpNotificationFormV2.tsx

```diff
@@ -0,0 +1,310 @@
+                         Undo Reset
+                       </Button>
+                     ) : undefined} />
+            )}
+          </Col>
+        </Row>
+        <Row>
+          <Col md={6}>
+            <Input id="api_key"
+                   name="api_key"
+                   label={<span>API Key <small className="text-muted">(Optional)</small></span>}
+                   type="text"
+                   onChange={this.handleChange}
+                   bsStyle={validation.errors.api_key ? 'error' : null}
+                   help={get(validation, 'errors.api_key[0]', 'Must be set if an API secret is set')}
+                   value={config.api_key} />
+          </Col>
+          <Col md={6}>
+            {api_secret?.keep_value ? (
+              <>
+                <ControlLabel>API Secret</ControlLabel>
+                <StyledButton bsStyle="default" type="button" onClick={() => { this.resetSecret('api_secret'); }}>
+                  Reset Secret
+                </StyledButton>
+              </>
+            ) : (
+              <Input id="apiSecret"
+                     label={<span>API Secret <small className="text-muted">(Optional)</small></span>}
+                     name="api_secret"
+                     type="password"
+                     onChange={this.handleSecretInputChange}
+                     bsStyle={validation.errors.api_secret ? 'error' : null}
+                     help={get(validation, 'errors.api_secret[0]', 'Must be set if an API key is set')}
+                     value={this.state.api_secret || ''}
+                     buttonAfter={reset.api_secret ? (
+                       <Button type="button" onClick={() => { this.undoResetSecret('api_secret'); }}>
+                         Undo Reset
+                       </Button>
+                     ) : undefined} />
+            )}
+          </Col>
+        </Row>
+        <Row>
+          <Col md={4}>
+            <Input help="HTTP method used for the notification"
+                   id="notification-method"
+                   label="HTTP Method">
+              <Select id="method"
+                      name="method"
+                      clearable={false}
+                      options={httpMethods}
+                      matchProp="label"
+                      onChange={this.handleMethodChange}
+                      value={config.method} />
+            </Input>
+          </Col>
+          <Col md={4}>
+            <Input help="HTTP content type used for POST/PUT notifications"
+                   id="notification-content-type"
+                   label="Content Type">
+              <Select id="content-type"
+                      name="content-type"
+                      options={contentTypes}
+                      matchProp="label"
+                      disabled={config.method === 'GET'}
+                      onChange={this.handleContentTypeChange}
+                      value={config.content_type} />
```

Comment:
Noticed you can clear the content type for POST and PUT which causes a very cryptic error.
```suggestion
                      clearable={false}
                      value={config.content_type} />
```